### PR TITLE
Update to newer `redis-py`

### PR DESCRIPTION
**Pull request by judahrand**
Signed-off-by: Judah Rand <17158624+judahrand@users.noreply.github.com>

**What this PR does / why we need it**:

As of `redis-py` 4.1 there is a cluster mode included which is based on an improved version of our old dependency. This change will remove a dependency and also allow us to use a newer version of the package.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

File: sdk/python/setup.py [hunk 1]
```diff
@@ -77,7 +77,7 @@
 ]
 
 REDIS_REQUIRED = [
-    "redis-py-cluster==2.1.2",
+    "redis==4.1.*",
```
Comment (adchia) [inline sdk/python/setup.py:80]: thoughts on this being too restrictive? ideally, we'd support both, but users might be using other versions of redis locally too.

**Comment by pyalex** [comment]
Hey @judahrand, can you please update requirements/*.txt files as well, so tests would pass?

**Comment by judahrand** [comment]
> Hey @judahrand, can you please update requirements/*.txt files as well, so tests would pass?

I'll do this but I was confused as I could not find the `*.in` files used to generate them with `pip-tools`. Am I missing them?

**Comment by judahrand** [comment]
> Hey @judahrand, can you please update requirements/*.txt files as well, so tests would pass?

Ah! Got it `make lock-python-dependencies`.

**Review by pyalex — APPROVED** [review]

File: sdk/python/setup.py (hunk 1 above)
Comment (adchia) [reply sdk/python/setup.py:80]: i guess we can do this now and see if anybody complains

Comment (adchia) [reply sdk/python/setup.py:80]: maybe at least do a redis >= 4.1.0?

File: sdk/python/setup.py [hunk 2]
```diff
@@ -101,15 +101,15 @@
...
+    "pytest>=6.0.0",
     "pytest-cov",
     "pytest-xdist",
     "pytest-benchmark>=3.4.1",
     "pytest-lazy-fixture==0.6.3",
     "pytest-timeout==1.4.2",
     "pytest-ordering==0.6.*",
     "pytest-mock==1.10.4",
-    "Sphinx!=4.0.0",
+    "Sphinx!=4.0.0,<4.4.0",
```
Comment (adchia) [inline sdk/python/setup.py:112]: was this Sphinx change intentional?

Comment (judahrand) [reply sdk/python/setup.py:112]: Yes - without this restriction `pip-tools` was unable to find a valid set of dependencies for Python 3.7 🤷

**Review by adchia — APPROVED** [review]

**Review by adchia — APPROVED** [review]
/lgtm

**Pull request merged**